modules/compute/gke-node-pool/reservation_definitions.tf

```diff
@@ -56,10 +56,11 @@ locals {
   verified_specific_reservations = [for k, v in data.google_compute_reservation.specific_reservations : v if(v.specific_reservation != null && v.specific_reservation_required == true)]
 
   # Build two maps to be used to compare the VM properties between reservations and the node pool
-  reservation_vm_properties = [for r in local.verified_specific_reservations : {
+  # Skip this for TPUs
+  reservation_vm_properties = local.has_gpu ? ([for r in local.verified_specific_reservations : {
```

Comment:
Please refactor to `for reservation in local.verified_specific_reservations`
Improves readability of the code